legate/core/runtime.py

```diff
@@ -2086,6 +2088,17 @@ def wrapper(*args: Any, **kwargs: Any) -> Any:
 
         return wrapper
 
+    def find_or_create_array_type(
+        self, elem_type: ty.Dtype, n: int
+    ) -> ty.Dtype:
+        key = (elem_type, n)
+        cached = self._array_type_cache.get(key)
+        if cached is not None:
+            return cached
+        arr_type = ty.array_type(elem_type, n)
+        self._array_type_cache[key] = arr_type
+        return arr_type
```

Comment:
This is another common way to express this kind of thing:
```suggestion
        key = (elem_type, n)
        if key not in self._array_type_cache:
            arr_type = ty.array_type(elem_type, n)
            self._array_type_cache[key] = arr_type       
        return self._array_type_cache[key]
```

Reply:
I remember @jjwilke, you, and I talked about this pattern when we were on #457 and the pattern you suggested seems slightly less efficient than my original code because it tries to look up `key` twice from the dict. Is that not the case? I actually like your suggestion and that's what I believe I've been writing, but since our discussion, I've been kind of converted, though I honestly don't care that much about this micro-optimization.

Reply:
It's definitely one more lookup, but its also less variable assignment and use, which also incur lookups into the namespace, more or less. I'd just be surprised if it didn't wash out (or wasn't acceptable even if it didn't quite) FWIW what I would typically write myself is actually
```
        key = (elem_type, n)
        if key not in self._array_type_cache:
            self._array_type_cache[key] = ty.array_type(elem_type, n)       
        return self._array_type_cache[key]
```